jetty-http/src/test/java/org/eclipse/jetty/http/HttpURITest.java

```diff
@@ -323,109 +327,134 @@ public void testCanonicalDecoded()
         return Arrays.stream(new Object[][]
             {
                 // Simple path example
-                {"http://host/path/info", "/path/info", EnumSet.noneOf(Ambiguous.class)},
-                {"//host/path/info", "/path/info", EnumSet.noneOf(Ambiguous.class)},
-                {"/path/info", "/path/info", EnumSet.noneOf(Ambiguous.class)},
+                {"http://host/path/info", "/path/info", EnumSet.noneOf(Violation.class)},
```

Comment:
Unrelated to UTF16, but we don't seem to have any tests where the scheme name contains a ".",  which is a legal character.
Or percent encoded host names?

Reply:
I have added some extra tests for this.